File: app/src/main/java/com/absinthe/kage/device/cmd/SetAudioModeCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.client.Client;

public class SetAudioModeCommand extends Command {

    public static final int MODE_ = 0;
    public int mode = MODE_;

    public SetAudioModeCommand() {
        cmd = IpMessageConst.MEDIA_SET_AUDIO_MODE;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .append(String.valueOf(mode))
                .build();
    }

    @Override
    public void doWork(Client client, String received) {

    }

    @Override
    public boolean parseReceived(String received) {
        return false;
    }
}

File: app/src/main/kotlin/com/absinthe/kage/device/cmd/SetAudioModeCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client

class SetAudioModeCommand : Command() {
    @JvmField
    var mode = MODE_

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .append(mode.toString())
                .build()
    }

    override fun doWork(client: Client, received: String) {}

    override fun parseReceived(received: String): Boolean {
        return false
    }

    companion object {
        const val MODE_ = 0
    }

    init {
        cmd = IpMessageConst.MEDIA_SET_AUDIO_MODE
    }
}